podpac/core/algorithm/reprojection.py

```diff
@@ -40,29 +43,44 @@ class Reproject(Interpolate):
                            the 'coordinates' attribute""",
     ).tag(attr=True)
 
+    reproject_dims = tl.List(trait=tl.Unicode(), allow_none=True, default_value=None).tag(attr=True)
+
     @tl.validate("coordinates")
     def _validate_coordinates(self, d):
-        if isinstance(d["value"], Node) and not hasattr(d["value"], "coordinates"):
-            raise ValueError(
-                "When specifying the coordinates as a PODPAC Node, this Node must have a 'coordinates' attribute"
-            )
-        return d["value"]
+        val = d["value"]
+        if isinstance(val, Node):
+            if not hasattr(val, "coordinates"):
+                raise ValueError(
+                    "When specifying the coordinates as a PODPAC Node, this Node must have a 'coordinates' attribute"
+                )
+        elif isinstance(val, dict):
+            Coordinates.from_definition(self.coordinates)
+        elif isinstance(val, string_types):
+            Coordinates.from_json(self.coordinates)
+        return val
 
-    @property
-    def _coordinates(self):
+    @cached_property
+    def reprojection_coordinates(self):
+        # get coordinates
         if isinstance(self.coordinates, Coordinates):
-            return self.coordinates
+            coordinates = self.coordinates
         elif isinstance(self.coordinates, Node):
-            return self.coordinates.coordinates
+            coordinates = self.coordinates.coordinates
         elif isinstance(self.coordinates, dict):
-            return Coordinates.from_definition(self.coordinates)
+            coordinates = Coordinates.from_definition(self.coordinates)
         elif isinstance(self.coordinates, string_types):
-            return Coordinates.from_json(self.coordinates)
-        else:
-            raise TypeError("The coordinates attribute is of the wrong type.")
+            coordinates = Coordinates.from_json(self.coordinates)
+
+        # drop non-reprojection dims
+        if self.reproject_dims is not None:
+            coordinates = coordinates.drop([dim for dim in coordinates if dim not in self.reproject_dims])
+
+        return coordinates
 
     def _source_eval(self, coordinates, selector, output=None):
-        return self.source.eval(self._coordinates, output=output, _selector=selector)
+        coords = self.reprojection_coordinates.intersect(coordinates, outer=True)
+        coords = merge_dims([coords, coordinates.drop(self.reproject_dims or self.reprojection_coordinates.dims)])
```

Comment:
Yes, this looks good. Also handles the case when the request has MORE coordinates than the reprojection coordinates. This is great. Thanks Jeff.